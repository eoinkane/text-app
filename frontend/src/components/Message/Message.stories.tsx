import React from "react";
import MessageComponent from "./index";
import Container from "@material-ui/core/Container";

export default {
  title: "Message",
  component: MessageComponent,
};

export const Message = () => {
  const text =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ";
  return (
    <Container style={{ backgroundColor: "#f5f5f5", padding: "4vh 4vw" }}>
      <MessageComponent text={text} />
    </Container>
  );
};
