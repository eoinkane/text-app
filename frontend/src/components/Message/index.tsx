import React from "react";
import cx from "clsx";
import { Paper, Typography } from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface MessagePropsBase {}

interface MessagePropsLeft extends MessagePropsBase {
  left: boolean;
}

interface MessagePropsRight extends MessagePropsBase {
  right: boolean;
}

type MessageProps = MessagePropsLeft | MessagePropsRight;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      maxWidth: "85%",
      padding: "10px 0",
    },
    left: {
      marginRight: "auto",
    },
    right: {
      marginLeft: "auto",
    },
    paper: {
      padding: "2vh 2vh",
      borderRadius: 10,
    },
  })
);

const Message: React.FC<MessageProps> = ({ children, ...rest }) => {
  const classes = useStyles();
  return (
    <div
      className={cx(
        classes.root,
        {
          [classes.left]:
            rest.hasOwnProperty("left") && (rest as MessagePropsLeft).left,
        },
        {
          [classes.right]:
            rest.hasOwnProperty("right") && (rest as MessagePropsRight).right,
        }
      )}
    >
      <Paper className={classes.paper} elevation={3}>
        <Typography>{children}</Typography>
      </Paper>
    </div>
  );
};

export default Message;
