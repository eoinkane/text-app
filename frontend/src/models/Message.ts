import axios, { AxiosResponse } from "axios";
import User from "./User";

interface MessageProps {
  id?: number;
  sender?: User;
  message?: string;
  conversationsId?: number;
}

type valueof<T> = T[keyof T];

export default class Message {
  constructor(private data: MessageProps) {}

  get<K extends keyof MessageProps>(propName: K): valueof<MessageProps> {
    return this.data[propName];
  }

  set(update: MessageProps): void {
    Object.assign(this.data, update);
  }
  fetch(): void {
    // axios
    //   .get(`http://localhost:3000/users/${this.get("id")}`)
    //   .then((response: AxiosResponse): void => {
    //     this.set(response.data);
    //   });
  }

  save(): void {
    // const id = this.get("id");
    // if (id) {
    //   axios.put(`http://localhost:3000/users/${id}`, this.data);
    // } else {
    //   axios.post(`http://localhost:3000/users`, this.data);
    // }
  }
}
