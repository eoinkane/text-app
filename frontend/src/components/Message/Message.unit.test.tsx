import React from "react";
import { shallow } from "enzyme";
import Message from "./index";

describe("Message", () => {
  const text =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ";

  it("Renders ", () => {
    const messageLeft = shallow(<Message left>{text}</Message>);
    const messageRight = shallow(<Message right>{text}</Message>);
    const messageCentre = shallow(
      <Message left right>
        {text}
      </Message>
    );

    expect(messageLeft).toMatchSnapshot();
    expect(messageRight).toMatchSnapshot();
    expect(messageCentre).toMatchSnapshot();
  });

  it("Renders with the correct text", () => {
    const message = shallow(<Message left>{text}</Message>);

    expect(message).toMatchSnapshot();
    expect(message.text()).toBe(text);
  });

  it("Renders at the left", () => {
    const message = shallow(<Message left>{text}</Message>);
    expect(message).toMatchSnapshot();
    expect(message.hasClass("makeStyles-left-2")).toBe(true);
  });

  it("Renders at the right", () => {
    const message = shallow(<Message right>{text}</Message>);
    expect(message).toMatchSnapshot();
    expect(message.hasClass("makeStyles-right-3")).toBe(true);
  });
});
