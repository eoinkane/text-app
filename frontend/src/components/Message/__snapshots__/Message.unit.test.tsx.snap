// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Renders  1`] = `
<div
  className="makeStyles-root-1 makeStyles-left-2"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Message Renders  2`] = `
<div
  className="makeStyles-root-1 makeStyles-right-3"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Message Renders  3`] = `
<div
  className="makeStyles-root-1 makeStyles-left-2 makeStyles-right-3"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Message Renders at the left 1`] = `
<div
  className="makeStyles-root-1 makeStyles-left-2"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Message Renders at the right 1`] = `
<div
  className="makeStyles-root-1 makeStyles-right-3"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`Message Renders with the correct text 1`] = `
<div
  className="makeStyles-root-1 makeStyles-left-2"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-4"
    elevation={3}
  >
    <WithStyles(ForwardRef(Typography))>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;
